agent/stats/statsretriever/stats_retriever_windows.go

```diff
@@ -0,0 +1,118 @@
+//go:build windows
+// +build windows
+
+// Copyright Amazon.com Inc. or its affiliates. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License"). You may
+// not use this file except in compliance with the License. A copy of the
+// License is located at
+//
+//	http://aws.amazon.com/apache2.0/
+//
+// or in the "license" file accompanying this file. This file is distributed
+// on an "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either
+// express or implied. See the License for the specific language governing
+// permissions and limitations under the License.
+
+package statsretriever
+
+import (
+	"syscall"
+	"unsafe"
+
+	log "github.com/cihub/seelog"
+	dockerstats "github.com/docker/docker/api/types"
+	"github.com/pkg/errors"
+	"golang.org/x/sys/windows"
+)
+
+type retriever struct {
+	// funcGetIfTable2Ex is the system call for GetIfTable2Ex Win32 API.
+	funcGetIfTable2Ex func(a ...uintptr) (r1 uintptr, r2 uintptr, lastErr error)
+	// funcFreeMibTable is the system call for FreeMibTable Win32 API.
+	funcFreeMibTable func(a ...uintptr) (r1 uintptr, r2 uintptr, lastErr error)
+}
+
+// NewStatsRetriever returns an instance of StatsRetriever interface.
+func NewStatsRetriever() StatsRetriever {
+	log.Debugf("Initializing stats retriever for Windows")
+	obj := &retriever{}
+
+	// We would be using GetIfTable2Ex and FreeMibTable Win32 APIs from IP Helper.
+	moduleIPHelper := windows.NewLazySystemDLL("iphlpapi.dll")
+	procGetIfTable2Ex := moduleIPHelper.NewProc("GetIfTable2Ex")
```

Comment:
Why are we retrieving the entire table and not the specific interface row?
https://learn.microsoft.com/en-us/windows/win32/api/netioapi/nf-netioapi-getifentry2ex?

Reply:
Ah we have been searching for this particular method all along. Unfortunately, we were not aware about the same. 
I'll make the changes to use `GetIfEntry2Ex` API.